cmd/kube-scheduler/app/server.go

```diff
@@ -158,12 +158,14 @@ func Run(ctx context.Context, cc *schedulerserverconfig.CompletedConfig, sched *
 	}
 
 	// Prepare the event broadcaster.
-	if cc.Broadcaster != nil && cc.EventClient != nil {
-		cc.Broadcaster.StartRecordingToSink(ctx.Done())
-	}
-	if cc.CoreBroadcaster != nil && cc.CoreEventClient != nil {
-		cc.CoreBroadcaster.StartRecordingToSink(&corev1.EventSinkImpl{Interface: cc.CoreEventClient.Events("")})
+	cc.EventClient.StartRecordingToSink(ctx.Done())
+
+	// If scheduler is using new Event API, we should manually start core broadcaster
+	// for leader election since it hasn't been migrated yet.
+	if cc.EventClient.V1beta1Client != nil {
```

Comment:
Let's make it part of the fallback in the library

Reply:
See my comment in the original PR.

Reply:
Updated the codes.